### DISCUSSION TITLE
Abot the new JSON export featue

### DISCUSSION THREAD

**Discussion by vonfritz** [ooflet/Mi-Create#42] [Questions-EN]
Hi,
 I saw that you started with the JSON export and I have
some questions about it. I saw that in the Jason export you export the Type device that is not mentioned in the GMF compiler documentation. Is this entry purely for informational purposes or do you have other information about it?
Another small annoying problem, this one with linux, when you create a new folder it is created with the name 'New girectory'. When you try to rename it the wording New directory does not disappear and the new name is inserted in the default one. Can't you make it so that when you start typing the new name the default one disappears?
Thanks

**Comment by vonfritz** [comment]
Another problem, when trying to export an "imageList" with "dataSrc" = None it gives an error.
This is necessary when you want to export an animation that does not have the "dataSrc" set.
So is it possible to make it not give an error and export the animation widget of type: anim_[xx@xx] to "type":"element_anim" ........?

**Comment by ooflet** [comment]
DeviceType is purely informational for Mi Create. Json projects dont specify what device they're made for, so to keep it simple for now and to avoid asking the user all the time what device the project was made for, a deviceType property was added to the exported json. You can safely remove this property if it interferes with something.

**Comment by vonfritz** [comment]
Thanks for the reply.
I noticed that there are other things that are not working, but maybe it is better to wait until it is implemented in the "main" branch.

**Comment by vonfritz** [comment]
I didn't wait for the integration in the main branch, because there were some errors that did not allow export for devices other than the mi band8. I took the liberty of making a pull request to be able to use the export with other devices as well.

**Reply by ooflet** [reply]
I apologise for the wait. Thank you for providing a fix.
